internal/app/cache/cache.go

```diff
@@ -0,0 +1,31 @@
+package cache
+
+import (
+	envoy_service_discovery_v3 "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
+)
+
+type Cache interface {
+	// Exists returns true if the key exists in the cache, false otherwise.
+	Exists(key string) bool
+
+	// Fetch returns the cached response if it exists.
+	Fetch(key string) (*envoy_service_discovery_v3.DiscoveryResponse, error)
+
+	// IsStreamOpen returns false if no stream is opened against the upstream control plane for the given key.
+	IsStreamOpen(key string) (bool, error)
```

Comment:
both  `IsStreamOpen` and `SetStreamOpen` related functions are probably internal to implementation. If the cache is a write through cache the g-c-p connection will be handled by the implementation.

Reply:
I agree (https://github.com/envoyproxy/xds-relay/pull/18#discussion_r391852472) - I removed both from the public interface.